circuit-foundation/src/commonMain/kotlin/com/slack/circuit/foundation/NavigableCircuitContent.kt

```diff
@@ -111,21 +112,17 @@ public fun <R : Record> NavigableCircuitContent(
   val outerKey = "_navigable_registry_${currentCompositeKeyHash.toString(MaxSupportedRadix)}"
   val outerRegistry = rememberRetained(key = outerKey) { RetainedStateRegistry() }
 
-  val saveableStateHolder = rememberSaveableStateHolder()
-
   CompositionLocalProvider(LocalRetainedStateRegistry provides outerRegistry) {
     decoration.DecoratedContent(activeContentProviders, backStack.size, modifier) { provider ->
       val record = provider.record
 
-      saveableStateHolder.SaveableStateProvider(record.key) {
```

Comment:
It looks like we're also no longer saving state during this section on top of disposing when we're done, what was the motivation for this?

Reply:
I didn't fully understand the question. The removed `saveableStateHolder` has been moved to `buildContentProviders`. The reason for this is that, similar to the `CanRetainChecker` in `buildContentProviders`, it determines whether a specific record exists in the backStack, so it seemed appropriate to place it in the same location.

From what I observed, as long as the record exists in the backStack, all values are correctly saved and restored. Is there any other concern I might have missed?

Reply:
The PR description said it just was disposing states at the end but this is also changing behavior, so I just wanted to understand better. Let me take another look later when I'm back at a computer

Reply:
Thanks! Feel free to let me know if I’ve implemented something incorrectly or missed anything.